ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKGroupedTable.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.structured;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.function.FunctionRegistry;
+import io.confluent.ksql.function.KsqlAggregateFunction;
+import io.confluent.ksql.function.KsqlUndoableAggregationFunction;
+import io.confluent.ksql.function.udaf.KudafAggregator;
+import io.confluent.ksql.function.udaf.KudafUndoAggregator;
+import io.confluent.ksql.parser.tree.WindowExpression;
+import io.confluent.ksql.util.KsqlException;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.streams.kstream.Initializer;
+import org.apache.kafka.streams.kstream.KGroupedTable;
+import org.apache.kafka.streams.kstream.KTable;
+import org.apache.kafka.streams.kstream.Materialized;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class SchemaKGroupedTable extends SchemaKGroupedStream {
```

Comment:
I agree we shouldn't use inheritance here. But we're kind of forced to because of how the other Schema.*(Stream|Table) classes are structured. SchemaKTable is a sub-class of SchemaKStream, and the plan nodes only ever deal with SchemaKStreams. So to keep things consistent I mirrored that here. We should probably rethink the relationships between those 4 classes and do a refactor, but I'm not sure that it makes sense to do so in _this_ PR.